numpy/polynomial/chebyshev.py

```diff
@@ -1886,6 +1887,55 @@ def chebroots(c):
     return r
 
 
+def chebinterp(func, deg, args=()):
+    """Return the Chebyshev series instance that interpolates a given function
+    at the Chebyshev points of the first kind over the interval [-1, 1].
+
+    Parameters
+    ----------
+    func : function
+        The function to be approximated. It must be a function of a single
+        variable of the form f(x,a,b,c...), where a,b,c... are extra arguments
+        that can be passed in the `args` parameter.
+    deg : int or 1-D array_like
+        Degree(s) of the resulting polynomial expressed in a Chebyshev basis. If
+        `deg` is a single integer all terms up to and including the `deg`'th
+        term are included in the interpolation.
+    args : tuple, optional
+        Extra arguments to be used in the function call.
+
+    Returns
+    -------
+    new_series : series
+        A series that represents the polynomial interpolation
+        to the function.
+
+    Examples
+    --------
+    >>> from numpy import tanh, pi
+    >>> import numpy.polynomial.chebyshev as C
+    >>> C.chebinterp(lambda x: tanh(x)+0.5, 8, domain=[0, 2*np.pi])
+    array([ 1.30198814,  0.3540514 , -0.25209078,  0.14032589, -0.05706644,
+            0.01226391,  0.00367748, -0.00455335])
+
+    .. versionadded:: 1.14.0
+
+    """
+    if deg <= 0:
+        raise ValueError("deg must be non-negative")
+    if type(deg) != int:
+        raise TypeError("deg must be an integer")
```

Comment:
I will use the check from chebvander() since, as charris pointed out, the check occurs there as well. It was his wish to include the check also in chebinterp().